src/passes/Precompute.cpp

```diff
@@ -271,17 +271,17 @@ struct Precompute
     if (Properties::isConstantExpression(curr) || curr->is<Nop>()) {
       return;
     }
-    // Until engines implement v128.const and we have SIMD-aware optimizations
-    // that can break large v128.const instructions into smaller consts and
-    // splats, do not try to precompute v128 expressions.
-    if (curr->type.isVector()) {
-      return;
+    // To optimise for size, we avoid expanding small constants into larger ones
+    // like i32x4.splat(1) -> v128.const i32x4 1 1 1 1
+    if (curr->type.isVector() && getPassOptions().shrinkLevel != 0) {
```

Comment:
However I'm not sure about `shrinkLevel != 0` limitation. A series of identical constants should compress well via gzip or brotli and be almost as good as a splat + const pair

Reply:
I think it's OK to precompute splats even when shrinking. Using SIMD at all is already increasing code size for performance, so we might as well improve the performance wherever SIMD is used.